pkgs/applications/networking/mullvad/libwg.nix

```diff
@@ -0,0 +1,35 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, mullvad
+}:
+buildGoModule {
+  pname = "libwg";
+
+  inherit (mullvad)
+    version
+    src
+    ;
+
+  sourceRoot = "source/wireguard/libwg";
+
+  vendorSha256 = "qvymWCdJ+GY90W/Fpdp+r1+mTq6O4LyN2Yw/PjKdFm0=";
+
+  # XXX: hack to make the ar archive go to the correct place
+  # This is necessary because passing `-o ...` to `ldflags` does not work
+  # (this doesn't get communicated everywhere in the chain, apparently, so
+  # `go` complains that it can't find an `a.out` file).
+  GOBIN = "${placeholder "out"}/lib";
+  ldflags = [ "-buildmode=c-archive" ];
```

Comment:
```suggestion
  ldflags = [ "-s" "-w" "-buildmode=c-archive" ];
```

Reply:
What do the `-s` and `-w` flags do?

Reply:
https://stackoverflow.com/questions/22267189/what-does-the-w-flag-mean-when-passed-in-via-the-ldflags-option-to-the-go-comman

> You will get the smallest binaries if you compile with -ldflags '-w -s'.
>
> The -w turns off DWARF debugging information: you will not be able to use gdb on the binary to look at specific functions or set breakpoints or get stack traces, because all the metadata gdb needs will not be included. You will also not be able to use other tools that depend on the information, like pprof profiling.
>
> The -s turns off generation of the Go symbol table: you will not be able to use go tool nm to list the symbols in the binary. strip -s is like passing -s to -ldflags but it doesn't strip quite as much. go tool nm might still work after strip -s. I am not completely sure.
>
> None of these — not -ldflags -w, not -ldflags -s, not strip -s — should affect the execution of the actual program. They only affect whether you can debug or analyze the program with other tools.

https://pkg.go.dev/cmd/link


lets add the explanation of them to
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/go.section.md#ldflags-var-go-ldflags

Reply:
Any reason why that's not a default in Nixpkgs (or is it, but only when ldflags is unset)?

Reply:
@zowoq do you happen to know why?

Reply:
I've looked at doing it a couple of times previously but it wasn't a good default to set. Sometimes caused weird issues or just outright breakage and was generally just a PITA for the amount of derivations that would need to opt out. Homebrew and a couple of other distros also looked at making it a default but decided against it.

I haven't tried rebuilding all of `nixpkgs` with it since 1.16, once we're on 1.18 (hopefully soon) I'll take another look at it.